src/common/util/userTimeout.ts

```diff
@@ -0,0 +1,137 @@
+import { arrayPatternMatch } from './arrayPatternMatch'
+import { isStringPatternMatch } from './stringPatternMatch'
+
+/**
+ * Default Session Idle Time
+ *
+ * If the timeout configuration has not been set, or the User has no roles ( although the
+ * User shouldn't reach this point if they have no roles associated with them ) the
+ * default session idle time will be used.
+ */
+
+export const DEFAULT_SESSION_TIMEOUT = {
+    idleModalDisplayTime: 10,
+    pattern:
+        'ERROR: NO-SESSION_TIMEOUT_SET. You need to set a DEFAULT Session Timeout for this application through' +
+        'the configuration file. ie. use the pattern ".", @see unit tests. The totalIdleTime will be set to a low value.',
+    totalIdleTime: 480,
+}
+
+/**
+ * Is Role Match
+ *
+ * Checks if a User's role, matches a specified Regular Expression.
+ *
+ * We use a Regular Expression so that we can set the default session timeout via configuration ie. '.', hence we used
+ * JS .match over .includes.
+ *
+ * There will be a different default session timeout per application, and different session timeouts per user groups
+ * hence setting it via configuration.
+ *
+ * @param role - 'pui-case-manager'
+ * @param pattern - 'case-manager' / 'pui-' / '.'
+ * @returns {boolean}
+ */
+export const isRoleMatch = (role: string, pattern: string): boolean => {
+    return isStringPatternMatch(role, pattern)
```

Comment:
any reason why we can't call these functions directly?

Reply:
As above. Otherwise we would miss on explaining to anyone who picks this code up the following comment:

/**
 * Is Role Match
 *
 * Checks if a User's role, matches a specified Regular Expression.
 *
 * We use a Regular Expression so that we can set the default session timeout via configuration ie. '.', hence we used
 * JS .match over .includes.
 *
 * There will be a different default session timeout per application, and different session timeouts per user groups
 * hence setting it via configuration.
 *
 * @param role - 'pui-case-manager'
 * @param pattern - 'case-manager' / 'pui-' / '.'
 * @returns {boolean}
 */

Therefore I proxy when I see a use for explaining something.